### DISCUSSION TITLE
Markdownload's future

### DISCUSSION THREAD

**Discussion by deathau** [deathau/markdownload#342] [Announcements]
As you've all probably noticed, I've been very lax in my updates recently, but I'm hoping to change that.  
I've just pushed version 3.4.0 to the various extension stores, which contains a bunch of fixes provided by the community (listed in the readme, or you can go look at all the pull requests). I'm very grateful for all of these updates.

But I really have to pull my finger out and get a version working with Manifest V3. So, I've started re-writing the entire thing from scratch. In the current version of the extension, most of the processing is done in a background page, but with Manifest V3 I have to use a service worker instead of a background page for Chrome, but not for Firefox... Basically, cross-browser is a little messy at the moment, but it's doable, and I have a kinda working version locally.

One major change I've implemented is opening the extension in a sidebar/side panel, rather than the popup. 
Here's what it looks like in Firefox:
![Screenshot 2024-08-23 at 11 57 36 AM](https://github.com/user-attachments/assets/91098639-1382-41d1-a1f9-2141edca3de9)
And in a chromium browser (Wavebox):
![Screenshot 2024-08-23 at 12 00 38 PM](https://github.com/user-attachments/assets/7c4a7c56-9879-4f8d-b8ce-c31c53547b64)

I kinda like this better, as you can keep it open and clip things from different websites to the same file before downloading, but it's very different to the current extension, so I wanted to get some opinions on if you guys like the direction or not.

I am concerned that Safari does not have a sidebar, but I've been thinking for ages about making a Mac-OS native "app" to handle image downloads (if you've installed Markdownload on Safari, you'll know it has a pointless app anyway because that's just how things work in Apple land). This could be made into a separate window which behaves like the sidebar does in other browsers, but I haven't so much as looked into that yet.

So, what do you think? Am I on the right track?

**Comment by netentity** [comment]
This looks amazing! Excited to see this continue to integrate into my writing workflows.

**Comment by ricketyboo** [comment]
Oops, just saw the comment about a complete rewrite here! I was just about to queue up a discussion asking what you thought about a large rewrite on a variety of things code style things like `var` vs `let`, `==` vs `===` to avoid type coercion, and possibly some unit tests, and maybe even some build stuff to enable typescript or esnext. Seems like you're on it; but if you need some help, shout out!

Loving the side bar in the screens above! Definitely a good direction!

**Reply by deathau** [reply]
I'm honestly not 100% sure why, but for this extension, I have an aversion to using build stuff like webpack. The submission process for Firefox in particular requires you to submit the un-built code separately for review, for example. There are other, in-my-own-head reasons, which probably wouldn't stand up to any form of scrutiny, but for now I'm going to stick with the no-build-system policy.

**Comment by SamejSpenser** [comment]
I'm really happy to hear that MarkDownload is making progress and evolving.

The side tabs instead of the current pop-ups seem way better; this way, I think I'll lose less of my manual edits before capturing, since I won't lose the already edited content if I accidentally tap or click outside it.

**Comment by FitzFrobozz** [comment]
Seems like a good fit and direction, and I really like the look of it.

What happens to an unsaved MarkDownload sidebar clipping if a person wants to double-check something else (using the sidebar) before fully saving the clipping? E.g. I use bookmarks and [Simple Tab Groups](https://addons.mozilla.org/en-US/firefox/addon/simple-tab-groups) in the Firefox sidebar and sometimes I need to refer to them before I finalize a MarkDownload clipping I've started.

**Reply by deathau** [reply]
What I'm playing with at the moment is that the data in the sidebar will persist as long as the sidebar remains open, meaning you should be able to switch tabs and windows without losing anything. It also opens up new use-cases, like clipping stuff from a bunch of different websites into the same file before downloading.
(I'm still working out the details on capturing images in this case, but I think it'll work)

**Reply by FitzFrobozz** [reply]
Perfect, thank you. Of course I'll take anything you throw at us as I've been a big fan. Best Obsidian clipper that I've seen, hands down.

**Comment by FitzFrobozz** [comment]
Oh, forgot to ask: since you are rewriting from scratch and changing certain things fundamentally, any interest in adding support for templater?

**Reply by deathau** [reply]
As in, the Obsidian plugin? What sort of support do you mean? Like syntax highlighting, or like using templater syntax?

**Reply by FitzFrobozz** [reply]
Yes, the plugin, sorry. I meant supporting the use of templater syntax so that templater itself could work its own magic on MarkDownload notes. I.e. user clips a page and the final note has all the features one would expect to see in a typical/standard templater-created note. (Hope that didn't just confuse the question even further, still waking up.)

**Reply by deathau** [reply]
While I like the idea in general, I'm not really into the idea of building and supporting my own templater clone. Looking at the docs there's a lot of Obsidian-specific stuff in there too that would make no sense to port over. I'd just be building a new template engine with similar syntax to templater.

On the other hand, you could always include templater syntax in your... template (words are hard) and trigger it in obsidian later. (If I'm understanding correctly, feel free to correct me)

**Reply by FitzFrobozz** [reply]
That's certainly understandable. While templater is regarded by many (including me) to be an essential plugin, it is by no means some kind of standard that deserves universal support (and I hope I didn't come across as seeing it as such). And yes, that suggestion is something I've had on my list of things to look into. And I have my workaround workflow as well. Anyways, thanks for the consideration and sorry for the momentary thread/issue derail!

**Reply by deathau** [reply]
No problems. This is probably the perfect place to take big swings like that 😅

**Comment by rsgranne** [comment]
I’m glad you are working on the extension, & I appreciate it. However, will there be a way to avoid opening a sidebar every time I use the extension? All I ever want to do is copy text as Markdown. I don’t need anything more complex than that. Thanks!

**Reply by deathau** [reply]
That's a good point and I'll keep it in mind.  
I plan to keep all the existing functionality (where possible) including context menus and keyboard shortcuts, I just have to make sure to keep in mind that the sidebar won't always need to open.

🤔 I wonder if I can change the functionality of the toolbar button based on user preference...

**Comment by gaufde** [comment]
This is very exciting! Would making the safari/macOS "native" app mean that Markdownload would work on iOS through share menus?

**Comment by thomastthai** [comment]
The sidebar layout looks cleaner, uses the horizontal screen space better, and does not block information underneath like a popup. Making it an option helps those who just want to keep their clipping use simple.

**Comment by seatrout** [comment]
I don't want to harass you, but I am waiting on tenterhooks for the new version, especially if it will manage discontinuous highlights. There is nothing so good as Markdownload for web clipping, at least with my workflow, and I've tried everything else :-)

**Comment by lj020326** [comment]
My latest chrome update wants to remove the existing version.
Eagerly awaiting the updated extension.  
any update?

**Reply by jmMeessen** [reply]
Create issue #369 for visibility.

**Reply by lj020326** [reply]
FWIW - With the realization a few features are still WIP, I was able to perform some level of basic feature testing successfully with the extension locally using the branch "v4-from-scratch" with minor enhancements.

I submitted a [PR](https://github.com/deathau/markdownload/pull/368) for that branch with the aforementioned minor enhancements that I had to make in order to get the extension to work for some basic features using my Chrome browser setup.  

BTW - The newly enhanced sidebar extension looked awesome when I was using/testing with it. :)

**Comment by abd3lraouf** [comment]
I found this https://chromewebstore.google.com/detail/marksnip-markdown-web-cli/kcbaglhfgbkjdnpeokaamjjkddempipm

**Reply by jmMeessen** [reply]
It states that "_MarkSnip is open source software licensed under the MIT license._" but I see nowhere the link to the code base on the Chrome plugin page. Did I miss something ?

**Reply by Dboingue** [reply]
Same question.   The introduction there appears to be careful and clear about this being a fork for this repository. How does one find the list of forks from the repository, the button here would create a fork. I will explore...

GitHub search on Marksnip as input gives https://github.com/oqzl/marksnip.  And that is a not a fork of this repository and not the code of the extension, AFAICT (can tell).

Google search, second page, (first page has only G web store result and softonic type). 
It does have this in results:
https://chrome-stats.com/d/kcbaglhfgbkjdnpeokaamjjkddempipm

But then I tried as google variant input suggestion:  "Marksnip" markdown web clipper github.
and got this. https://napkforpc.com/crx/kcbaglhfgbkjdnpeokaamjjkddempipm/

 It seems to be a CRX package. Is that equivalent to source code when uncompressed?  (yes learning here, in passing).
It has more information than the Google web store, I would not know how to search further. Also single result google search (rare).

I hope this help others to find out what is going on there.

**Reply by jmMeessen** [reply]
FWIW, It appears that there is a fork of this project that seems to match the name markdown snip at https://github.com/DhruvParikh1/markdownload-extension-updated.

**Comment by jmMeessen** [comment]
(I always have to look it up too)
The list of forks can be viewed in the "insight" menu.

I browsed the list and couldn't find one approaching the new Chrome plugin.
I sent an email to the author for a pointer to the sources but had no reply
yet.

Hope this helps.

 I really miss the tool's functionality (using it here in conjunction with
Obsidian). The planned new major release looks very promissing and cross
browser.

**Reply by lj020326** [reply]
FWIW - With the realization a few features are still WIP, I was able to perform some level of basic feature testing successfully with the extension locally using the branch "v4-from-scratch" with minor enhancements.

I submitted a [PR](https://github.com/deathau/markdownload/pull/368) for that branch with the aforementioned minor enhancements that I had to make in order to get the extension to work for some basic features using my Chrome browser setup.  

BTW - The newly enhanced sidebar extension looked awesome when I was using/testing with it. :)

**Comment by mildlydiverting** [comment]
Expressing my appreciation for this extension - thank you! - and _gently encouraging_ you towards a release of a new version that will work with Chrome.

**Reply by gengjiawen** [reply]
I published a fork: https://chromewebstore.google.com/detail/modern-markdownload-markd/lfihmpgmbpingelkgmodjgghdcjeiikk?authuser=0&hl=en

**Reply by gauthierae** [reply]
> Expressing my appreciation for this extension - thank you! - and _gently encouraging_ you towards a release of a new version that will work with Chrome.

I've been following this discussion — I was a MarkDownload user too, and the MV2 deprecation was what pushed me to build something.

I made a Chrome extension called Atomic Clipper that covers most of what MarkDownload did with one key difference: instead of one-click full-page to Markdown, it's a DevTools-style element picker — you hover over any section of a page and clip exactly that part. The Markdown output preserves headings, code blocks, links, and lists.

Alternatively, you can use the "save article" mode that will save the main content of the page.

It also has a built-in local library where clips accumulate by category, with per-category .md export. I use this to feed research batches into NotebookLM, but the exports work anywhere you'd paste Markdown.

MIT licensed, open source, local-only: https://github.com/gauthierae/AtomicClipper/blob/main/README.md

This is an early version and I am looking for feedback.

Installing from GitHub (developer mode) — the README has the 5 steps. Happy to answer questions about how it compares to MarkDownload or discuss features.
